packages/components/src/color-picker/test/index.tsx

```diff
@@ -73,99 +38,188 @@ const legacyColorMatcher = {
 describe( 'ColorPicker', () => {
 	describe( 'legacy props', () => {
 		it( 'should fire onChangeComplete with the legacy color format', async () => {
+			const user = userEvent.setup();
 			const onChangeComplete = jest.fn();
-			const color = '#fff';
+			const color = '#000';
 
-			const { container } = render(
+			render(
 				<ColorPicker
 					onChangeComplete={ onChangeComplete }
 					color={ color }
+					enableAlpha={ false }
 				/>
 			);
 
-			const saturation = getSaturation( container );
+			const formatSelector = screen.getByRole( 'combobox' );
 
-			if ( saturation === null ) {
-				throw new Error( 'The saturation slider could not be found' );
+			if ( formatSelector === null ) {
```

Comment:
This check is redundant: `getByRole` itself will throw an error if it can't find the element. The previous `getSaturation` used `querySelector`, which doesn't throw like the `get*` query methods.

Reply:
I generally find throwing errors in tests odd. For any thrown error in a test, I'd expect an assertion to fail instead.

Reply:
Thank you both! This check (and the others like it elsewhere in the file) were initially [implemented](https://github.com/WordPress/gutenberg/pull/49214#discussion_r1142625772) when `querySelector` was in play. I just kind of carried it forward from there, but you're right it isn't needed with `getByRole`!